File: FlowMeter/classes/ui/LaunchViewController.m
//
//  LaunchViewController.m
//  FlowMeter
//
//  Created by Simon Bogutzky on 23.01.15.
//  Copyright (c) 2015 Simon Bogutzky. All rights reserved.
//

#import "LaunchViewController.h"

@interface LaunchViewController ()

@end

@implementation LaunchViewController

- (void)viewDidLoad {
    [super viewDidLoad];
    
    self.navigationController.navigationBarHidden = YES;
    [self performSelector:@selector(showTabBar) withObject:nil afterDelay:3.0];
}

- (BOOL)shouldAutorotate
{
    return NO;
}

- (void)showTabBar
{
    UIStoryboard *storyboard = [UIStoryboard storyboardWithName:@"MainStoryboard" bundle:nil];
    UITabBarController *tabBarController = [storyboard instantiateViewControllerWithIdentifier:@"TabBar"];
    tabBarController.selectedIndex = 1;
    tabBarController.tabBar.translucent = NO;
    
    [self presentViewController:tabBarController animated:NO completion:^{
        [UINavigationBar appearance].tintColor = [UIColor whiteColor];
        [UIApplication sharedApplication].statusBarStyle = UIStatusBarStyleLightContent;
        [UIApplication sharedApplication].statusBarHidden = NO;
    }];
}

@end

File: FlowMeter/classes/ui/LaunchViewController.swift
//
//  LaunchViewController.swift
//  FlowMeter
//
//  Created by Simon Bogutzky on 06.12.14.
//  Copyright (c) 2014 Simon Bogutzky. All rights reserved.
//

import UIKit

class LaunchViewController: UIViewController {

    override func viewDidLoad() {
        super.viewDidLoad()
        
        self.navigationController?.navigationBarHidden = true;
        
        var timer = NSTimer.scheduledTimerWithTimeInterval(3.0, target: self, selector: Selector("showTabBar"), userInfo: nil, repeats: false)
    }
    
    override func shouldAutorotate() -> Bool {
        return false
    }
    
    func showTabBar() {
        let storyboard = UIStoryboard(name: "MainStoryboard", bundle: nil)
        let tabBarController = storyboard.instantiateViewControllerWithIdentifier("TabBar") as UITabBarController
        
        tabBarController.selectedIndex = 1
        tabBarController.tabBar.translucent = false
        self.presentViewController(tabBarController, animated: false) { () -> Void in
            
            // Global apperance
            UINavigationBar.appearance().tintColor = UIColor.whiteColor()
            UIApplication.sharedApplication().statusBarStyle = .LightContent
            UIApplication.sharedApplication().statusBarHidden = false
        }
            
    }

}
